bookserver/crud.py

```diff
@@ -494,3 +580,189 @@ async def create_user_chapter_progress_entry(
     async with async_session.begin() as session:
         session.add(new_ucp)
     return UserChapterProgressValidator.from_orm(new_ucp)
+
+
+#
+# Select Question Support
+# -----------------------
+
+
+async def create_selected_question(
+    sid: str,
+    selector_id: str,
+    selected_id: str,
+    points: Optional[int] = None,
+    competency: Optional[str] = None,
+) -> SelectedQuestionValidator:
+    new_sqv = SelectedQuestion(
+        sid=sid,
+        selector_id=selector_id,
+        selected_id=selected_id,
+        points=points,
+        competency=competency,
+    )
+    async with async_session.begin() as session:
+        session.add(new_sqv)
+    return SelectedQuestionValidator.from_orm(new_sqv)
+
+
+async def fetch_selected_question(sid: str, selector_id) -> SelectedQuestionValidator:
+    query = select(SelectedQuestion).where(
+        (SelectedQuestion.sid == sid) & (SelectedQuestion.selector_id == selector_id)
+    )
+
+    async with async_session() as session:
+        res = await session.execute(query)
+        rslogger.debug(f"{res=}")
+        return SelectedQuestionValidator.from_orm(res.scalars().first())
+
+
+async def update_selected_question(sid: str, selector_id: str, selected_id: str):
+    stmt = (
+        update(SelectedQuestion)
+        .where(
+            (SelectedQuestion.sid == sid)
+            & (SelectedQuestion.selector_id == selector_id)
+        )
+        .values(selected_id=selected_id)
+    )
+    async with async_session.begin() as session:
+        await session.execute(stmt)
+    rslogger.debug("SUCCESS")
+
+
+# Questions and Assignments
+# -------------------------
+
+
+async def fetch_question(
+    name: str, basecourse: Optional[str] = None
+) -> QuestionValidator:
+
+    where_clause = Question.name == name
+    if basecourse:
+        where_clause = where_clause & (Question.base_course == basecourse)
+
+    query = select(Question).where(where_clause)
+
+    async with async_session() as session:
+        res = await session.execute(query)
+        rslogger.debug(f"{res=}")
+        return QuestionValidator.from_orm(res.scalars().first())
+
+
+auto_gradable_q = [
```

Comment:
From the old server,

``` python
AUTOGRADEABLE = set(
    [
        "clickablearea",
        "dragndrop",
        "fillintheblank",
        "khanex",
        "mchoice",
        "parsonsprob",
        "quizly",
        "selectquestion",
    ]
)
```

LP is also autogradable. What about activecode?

Reply:
Activecodes are not autogradable unless they have unit tests.

I don't know enough abuot LP questions to know whether they should be in this list or not.

Reply:
I don't understand why this comment is here it doesn't seem to fit.

Reply:
I see. All LP questions must have an associated unit test, so yes -- they are all autogradable.